p2p/discover/udp.go

```diff
@@ -529,9 +521,6 @@ func (req *ping) handle(t *udp, from *net.UDPAddr, fromID NodeID, mac []byte) er
 	if expired(req.Expiration) {
 		return errExpired
 	}
-	if req.Version != Version {
-		return errBadVersion
-	}
```

Comment:
A new version doesn't necessarily mean that the packet format changes.
This change can help in the following case:

We want to make a breaking change to rlpx. Nodes that have the new code need to know
the version of the node they're connecting in order to decide whether to use the old or the new
logic.